qt/place_panel.cpp

```diff
@@ -58,12 +57,64 @@ class QHLine : public QFrame
   }
 };
 
-PlacePageDialogUser::PlacePageDialogUser(QWidget * parent, place_page::Info const & info,
-                                         search::ReverseGeocoder::Address const & address)
-  : QDialog(parent)
+void PlacePanel::addCommonButtons(QDialogButtonBox * dbb, place_page::Info const & info)
+{
+  dbb->setCenterButtons(true);
+
+  QPushButton * fromButton = new QPushButton("From");
+  fromButton->setIcon(QIcon(":/navig64/point-start.png"));
+  fromButton->setAutoDefault(false);
+  connect(fromButton, &QAbstractButton::clicked, this, [this, info](){ routeFrom(info); });
+  dbb->addButton(fromButton, QDialogButtonBox::ActionRole);
+
+  QPushButton * addStopButton = new QPushButton("Stop");
+  addStopButton->setIcon(QIcon(":/navig64/point-intermediate.png"));
+  addStopButton->setAutoDefault(false);
+  connect(addStopButton, &QAbstractButton::clicked, this, [this, info](){ addStop(info); });
+  dbb->addButton(addStopButton, QDialogButtonBox::ActionRole);
+
+  QPushButton * routeToButton = new QPushButton("To");
+  routeToButton->setIcon(QIcon(":/navig64/point-finish.png"));
+  routeToButton->setAutoDefault(false);
+  connect(routeToButton, &QAbstractButton::clicked, this, [this, info](){ routeTo(info); });
+  dbb->addButton(routeToButton, QDialogButtonBox::ActionRole);
+
+  if (info.ShouldShowEditPlace())
+  {
+    QPushButton * editButton = new QPushButton("Edit Place");
+    connect(editButton, &QAbstractButton::clicked, this, [this, info](){ editPlace(info); });
+    dbb->addButton(editButton, QDialogButtonBox::AcceptRole);
+  }
+}
+
+PlacePanel::PlacePanel(QWidget * parent)
+  : QWidget(parent)
+{
+  setMaximumWidth(kMaximumPanelWidth);
+  setMinimumWidth(kMinimumPanelWidth);
+}
+
+void PlacePanel::setPlace(
+  place_page::Info const & info,
+  search::ReverseGeocoder::Address const & address
+)
```

Comment:
Does it fit 120 chars limit from our guideline?
```suggestion
void PlacePanel::setPlace(place_page::Info const & info, search::ReverseGeocoder::Address const & address)
```